power_grid_model_c/power_grid_model/include/power_grid_model/main_core/update.hpp

```diff
@@ -130,4 +135,194 @@ inline void update_inverse(MainModelState<ComponentContainer> const& state, Forw
                                      get_component_sequence<Component>(state, begin, end));
 }
 
+namespace update_independence {
```

Comment:
I would instead argue for a different name here
```suggestion
namespace update {
```

Reply:
I thought about `update` as well, but then the other functions in the file may as well go in there and I wanted to differentiate them. What do you think about that? Do you think renaming as `update` is good if we keep the function locations as is?

Reply:
In that case, I would think that moving the following three functions to live outside the namespaces is better:  `get_component_sequence`, `get_all_sequence_idx_map` x2.

Reply:
I actually really like this. `get_component_sequence` from `update.hpp` is actually now only used in `get_all_sequence_idx_map`. So I am going to move all flavors of `get_component_sequence` (and try to remove some if possible) into `namespace detail` and leave `get_all_sequence_idx_map` in the main `namespace update`. That way we also remove the ambiguity from `get_component_sequence` from `state_queries.hpp` which, on the contrary, is used across the whole codebase. In that sense, `namespace update_independence` can safely be renamed to `update`. I'll tag the commit later here.

Reply:
See [67929c5](https://github.com/PowerGridModel/power-grid-model/pull/828/commits/67929c5d2e9879f28f75620ce7de412ecaa0ce92)